pkg/reconciler/buildrun/resources/credentials.go

```diff
@@ -37,6 +37,15 @@ func ApplyCredentials(ctx context.Context, build *buildv1alpha1.Build, serviceAc
 		modified = updateServiceAccountIfSecretNotLinked(ctx, sourceSecret, serviceAccount) || modified
 	}
 
+	// if output is overridden by buildrun, and if this override has credentials,
+	// it should be added to the sa as well
+	if buildRun.Spec.Output != nil {
+		sourceSecret = buildRun.Spec.Output.Credentials
+		if sourceSecret != nil {
+			modified = updateServiceAccountIfSecretNotLinked(ctx, sourceSecret, serviceAccount) || modified
+		}
+	}
```

Comment:
I think if the BuildRun specifies a output credential, then the Build's credential (in case it is different) should not be added to the service account. -> You would need to move the section before your new block into an `else` of this block.

Or am I overseeing a use case?

Reply:
In my opinion it doesn't really matter. `BuildRun` override is most likely a one-time thing, otherwise it would be changed in a `Build` itself, and in that case secret is already linked to the service account.
I'll change it though, thanks.

Reply:
@SaschaSchwarze0 done